docs/contributing/contributing-rules.md

```diff
@@ -1,102 +1,316 @@
 
 ### Rule messages
 
-A well-written rule message includes:
+Include a rule message that provides details about the matched pattern and informs how to mitigate any related issues. Provide the following information in a rule message:
 
 1. Description of the pattern. For example: missing parameter, dangerous flag, out-of-order function calls.
 2. Description of why this pattern was detected. For example: logic bug, introduces a security vulnerability, bad practice.
 3. An alternative that resolves the issue. For example: Use another function, validate data first, and discard the dangerous flag.
 
-For an example of a good rule message, see: [this rule for Django's mark_safe](https://semgrep.dev/r?q=python.django.security.audit.avoid-mark-safe.avoid-mark-safe).
-
 Use the YAML multiline string operator `>-` when rule messages span multiple lines. This presents the best-looking rule message on the command-line without having to worry about line wrapping or escaping the quote or backslash characters.
 
-:::info
+For an example of a good rule message, see: [this rule for Django's mark_safe](https://semgrep.dev/r?q=python.django.security.audit.avoid-mark-safe.avoid-mark-safe).
+
+:::info Rule message example
 `mark_safe()` is used to mark a string as *safe* for HTML output. This disables escaping and may expose the content to XSS attacks. Instead, use `django.utils.html.format_html()` to build HTML for rendering.
 :::
 
-### Rule quality checker
+### Including additional details with rule metadatazs
+
+Rules in the Semgrep Registry require specific metadata fields that ensure consistency across the ecosystem in both Semgrep App and Semgrep CLI. These fields help users of Semgrep to identify rules in different categories such as:
+
+- High confidence security rules for CI pipelines.
+- OWASP Top 10 or CWE Top 25 rule packs.
+- Technology. For example `react` so it is easy to find `p/react` rule packs.
+- Audit rules with lower confidence intended for code auditors
+
+To help identify rules in these categories, the following list of metadata is required that enable the above categories:
+
+- cwe
+- references
+- category
+- subcategory
+- likelihood
+- impact
+- confidence
+
+An example of what this looks like can be found below:
+
+- [High confidence JavaScript/TypeScript rule](https://semgrep.dev/playground/r/javascript.express.security.audit.express-open-redirect.express-open-redirect)
+- [Medium confidence Python rule](https://semgrep.dev/playground/r/python.lang.security.dangerous-system-call.dangerous-system-call)
+- [Low confidence C# rule](https://semgrep.dev/playground/r/cs.lang.security.insecure-deserialization.fast-json.insecure-fastjson-deserialization)
```

Comment:
@LewisArdern this rule does not include all of the required metadata fields we show in this document. Should we use another as an example here?

Reply:
yes they do?

Reply:
![Screenshot 2022-11-09 at 16 27 11](https://user-images.githubusercontent.com/26669307/200871128-a0799235-d5e0-43db-8735-54acae9c4d63.png)